src/hisp/plamsa_data_handling/main.py

```diff
@@ -20,69 +22,82 @@ def __init__(
                 t_rel = t - t_pulse_start where t_pulse_start is the start of the pulse in seconds
             ion (bool, optional): _description_. Defaults to True.
 
         Returns:
             float: particle flux in part/m2/s
         """
+        if isinstance(nb_bin, SubBin):
+            bin_index = nb_bin.parent_bin_index
+            wetted_frac = nb_bin.wetted_frac
+        elif isinstance(nb_bin, DivBin):
+            bin_index = nb_bin.index
+            wetted_frac = 1
+
         if ion:
             FP_index = 2
             other_index = 0
+            flux_frac = wetted_frac
         if not ion:
             FP_index = 3
             other_index = 1
+            flux_frac = 1
 
         if pulse_type == "FP":
-            flux = self.pulse_type_to_data[pulse_type][:, FP_index][nb_mb - 1]
+            flux = self.pulse_type_to_data[pulse_type][:, FP_index][bin_index]
         elif pulse_type == "RISP":
             assert isinstance(
                 t_rel, float
             ), f"t_rel should be a float, not {type(t_rel)}"
-            flux = self.RISP_data(nb_mb=nb_mb, t_rel=t_rel)[other_index]
+            flux = self.RISP_data(bin=nb_bin, t_rel=t_rel)[other_index]
         elif pulse_type == "BAKE":
             flux = 0.0
         else:
-            flux = self.pulse_type_to_data[pulse_type][:, other_index][nb_mb - 1]
+            flux = self.pulse_type_to_data[pulse_type][:, other_index][bin_index]
 
-        return flux
+        return flux * flux_frac
 
-    def RISP_data(self, nb_mb: int, t_rel: float | int) -> NDArray:
-        """Returns the correct RISP data file for indicated monoblock
+    def RISP_data(self, bin: SubBin | DivBin, t_rel: float | int) -> NDArray:
+        """Returns the correct RISP data file for indicated bin
 
         Args:
-            nb_mb: mb number
+            bin: Subbin or Divbin object
             t_rel: t_rel as an integer(in seconds).
                 t_rel = t - t_pulse_start where t_pulse_start is the start of the pulse in seconds
 
         Returns:
             data: data from correct file as a numpy array
         """
-        inner_swept_bins = list(range(46, 65))
-        outer_swept_bins = list(range(19, 34))
-
-        if nb_mb in inner_swept_bins:
-            folder = self.path_to_RISP_data
-            div = True
-            offset_mb = 46
-        elif nb_mb in outer_swept_bins:
-            folder = self.path_to_ROSP_data
+        print(isinstance(bin, SubBin))
+        print(bin)
+        if isinstance(bin, SubBin):
+            bin_index = bin.parent_bin_index
+            div = False
+        elif isinstance(bin, DivBin):
+            bin_index = bin.index
             div = True
-            offset_mb = 19
+
+        if div:
+            if bin.inner_bin:
+                folder = self.path_to_RISP_data
+                offset_mb = 45
+            elif bin.outer_bin:
+                folder = self.path_to_ROSP_data
+                offset_mb = 18
```

Comment:
these offset_mb are still extremely dangerous imo. We need to think of a way to change this.